File: forminputjava/src/main/java/com/omarshehe/forminputjava/FormInputLayoutContract.java
package com.omarshehe.forminputjava;

import java.util.ArrayList;

public interface FormInputLayoutContract {

    public interface View {

    }

    public interface Presenter {
        public ArrayList<String> getTimeSlots();

        public String pad(int input);

        public boolean isValidEmail(CharSequence email);

        public boolean isValidPhoneNumber(CharSequence phoneNumber);
    }
}

File: forminputkotlin/src/main/java/com/omarshehe/forminputkotlin/FormInputLayoutContract.kt
package com.omarshehe.forminputkotlin

import java.util.ArrayList

interface FormInputLayoutContract {
    interface View
    interface Presenter {
        val timeSlots: ArrayList<String>

        fun pad(input: Int): String

        fun isValidEmail(email: CharSequence): Boolean

        fun isValidPhoneNumber(phoneNumber: CharSequence): Boolean
    }
}